trulens_eval/trulens_eval/feedback.py

```diff
@@ -1,5 +1,353 @@
+    ...
+    # The wrapped app.
+    app: Any = Field(exclude=True)
+    ...
+```
+
+The other non-excluded fields accessible outside of the wrapped app are listed
+in the `AppDefinition` class in `schema.py`:
+
+```python
+class AppDefinition(SerialModel, WithClassInfo, ABC):
+    ...
+
+    app_id: AppID
+
+    feedback_definitions: Sequence[FeedbackDefinition] = []
+
+    feedback_mode: FeedbackMode = FeedbackMode.WITH_APP_THREAD
+
+    root_class: Class
+
+    root_callable: ClassVar[FunctionOrMethod]
+
+    app: JSON
+```
+
+Note that `app` is in both classes. This distinction between `App` and
+`AppDefinition` here is that one corresponds to potentially non-serializable
+python objects (`App`) and their serializable versions (`AppDefinition`).
+Feedbacks should expect to be run with `AppDefinition`, fields of `App` that are
+not part of `AppDefinition` may not be available.
+
+Note that you can inspect the data available for feedback definitions in the
+dashboard by clicking on the "See full app json" button on the bottom of the
+page after selecting a record from a table.
+
+The other piece of context to Feedback evaluation are records. These contain the
+inputs/outputs and other information collected during the execution of an app:
+
+```python
+class Record(SerialModel):
+    record_id: RecordID
+    app_id: AppID
+
+    cost: Optional[Cost] = None
+    perf: Optional[Perf] = None
+
+    ts: datetime = pydantic.Field(default_factory=lambda: datetime.now())
+
+    tags: str = ""
+
+    main_input: Optional[JSON] = None
+    main_output: Optional[JSON] = None  # if no error
+    main_error: Optional[JSON] = None  # if error
+
+    # The collection of calls recorded. Note that these can be converted into a
+    # json structure with the same paths as the app that generated this record
+    # via `layout_calls_as_app`.
+    calls: Sequence[RecordAppCall] = []
+```
+
+A listing of a record can be seen in the dashboard by clicking the "see full
+record json" button on the bottom of the page after selecting a record from the
+table.
+
+##### App-organized Calls
+
+When evaluating a feedback function, Records are augmented with
+app/component calls in app layout in the attribute `app`. By this we mean that
+in addition to the fields listed in the class definition above, the `app` field
+will contain the same information as `calls` but organized in a manner mirroring
+the organization of the app structure. For example, if the instrumented app
+contains a component `combine_docs_chain` then `app.combine_docs_chain` will
+contain calls to methods of this component. In the example at the top of this
+docstring, `_call` was an example of such a method. Thus
+`app.combine_docs_chain._call` further contains a `RecordAppCall` (see
+schema.py) structure with information about the inputs/outputs/metadata
+regarding the `_call` call to that component. Selecting this information is the
+reason behind the `Select.RecordCalls` alias (see next section).
```

Comment:
Maybe move these lines towards the bottom (given we should get to the selectors portion sooner)

Add to a section

## Additional App Fields supplied by TruLens

Reply:
reordered last sections